float64 x
float64 z
float64 diameter
float64 diam_major
float64 diam_minor
float64 cov_angle
